src/driver/sap/SapQueryRunner.ts

```diff
@@ -1167,13 +1167,52 @@ export class SapQueryRunner extends BaseQueryRunner implements QueryRunner {
                 `Column "${oldTableColumnOrName}" was not found in the "${table.name}" table.`,
             )
 
-        if (
+        // Check if this is a safe length increase for compatible types
+        const isSafeLengthIncrease =
+            oldColumn.type.toLowerCase() === newColumn.type.toLowerCase() &&
+            oldColumn.length != null &&
+            newColumn.length != null &&
+            Number(newColumn.length) > Number(oldColumn.length) &&
+            (newColumn.type.toLowerCase() === "varchar" ||
+                newColumn.type.toLowerCase() === "char" ||
+                newColumn.type.toLowerCase() === "nvarchar" ||
+                newColumn.type.toLowerCase() === "nchar") &&
+            oldColumn.isGenerated === newColumn.isGenerated
+
+        if (isSafeLengthIncrease) {
+            // Use ALTER COLUMN for safe length increases
+            const lengthPart = newColumn.length ? `(${newColumn.length})` : ""
+            upQueries.push(
+                new Query(
+                    `ALTER TABLE ${this.escapePath(table)} ALTER ("${
+                        oldColumn.name
+                    }" ${newColumn.type.toUpperCase()}${lengthPart})`,
+                ),
+            )
+            downQueries.push(
+                new Query(
+                    `ALTER TABLE ${this.escapePath(table)} ALTER ("${
```

Comment:
The up query is safe, but not the down query.

The following scenario will fail:
1. run the migration that increases length from 50 to 100
2. save some data into the table
3. revert the migration - fails.

What we do in our project is this:
```sql
RENAME COLUMN "table_name"."column_name" TO "column_name_typeorm_backup";
ALTER TABLE "table_name" ADD ("column_name" nvarchar(50));
UPDATE "table_name" SET "column_name" = LEFT("column_name_typeorm_backup", 50) WHERE LENGTH("column_name_typeorm_backup") > 50;
ALTER TABLE "table_name" DROP ("column_name_typeorm_backup");
```

We could also do the same for the up query when the length decreases 😉 Still better than dropping the column 🤷‍♂